tools/cli/commands/changelog.js

```diff
@@ -390,16 +412,59 @@ async function changelogArgs( argv ) {
 				argv.args.push( argv.ver );
 			}
 			break;
+		case 'squash':
+			if ( argv.r && argv.readme ) {
+				readme = true;
+			} else {
+				readme = await promptReadme( argv );
+			}
+			argv.args = [ 'squash' ];
+			break;
 	}
 
-	// Remove project from arguments list we pass to the changelogger.
+	// Remove the project from the list of args we're passing to changelogger.
+	argv = await removeArgs( argv, removeArg );
+
+	// Run the changelogger command.
+	await changeloggerCli( argv );
+
+	// Squash just the readme if necessary.
+	if ( readme ) {
+		readmeSquash( argv );
+	}
+
+	// Add any newly added changelog files.
+	await gitAdd( argv );
+
+	console.log( chalkJetpackGreen( argv.success ) );
+}
+
+/**
+ * Handles squashing just the readme file.
+ *
+ * @param {object} argv - arguments passed as cli.
+ */
+async function readmeSquash( argv ) {
+	console.log( 'Updating readme...' );
+	await runCommand( 'tools/plugin-changelog-to-readme.sh', `${ argv.project }` );
+	console.log( 'Reverting changelog' );
+	await runCommand( 'git', [ 'checkout', '--', `projects/${ argv.project }/CHANGELOG.md` ] );
```

Comment:
I note this doesn't get called if something goes wrong with the readme update, which will leave the squashed CHANGELOG.md in place.

I also note that if there were any uncommitted changes to CHANGELOG.md, this will wipe them out.

Overall, it might be better to read CHANGELOG.md into a local variable before running changelogger, then write it back out instead of doing a `git checkout`. Something along the lines of
```js
const changelogContents = readme ? fs.readFileSync(  `projects/${ argv.project }/CHANGELOG.md` ) : null;
try {
    await changeloggerCli( argv );

    if ( argv.args[0] === 'squash' ) { // or `readme` depending on which way you go with the comment on line 229
        console.log( 'Updating readme...' );
        await runCommand( 'tools/plugin-changelog-to-readme.sh', [ `${ argv.project }` ] );
    }
} finally {
    if ( changelogContents !== null ) {
        fs.writeFileSync( `projects/${ argv.project }/CHANGELOG.md`, changelogContents );
    }
}

Reply:
Took a shot at rolling this and the other comment all into one - instead of taking readme as a boolean, we just ask if you want to squash the changelog or the readme, which seems cleaner? 

I think some things might still be broken, not sure what's expected. When I try to run, I get:

When running the option with changelog: 

> Only a single entry matched, not squashing

But it seems to squash things anyway (or try to), If I just attempt to run the `readme`, or if I run the script itself, same behavior but without the error.

Reply:
Looks like you're calling changelogger twice, once from line 422 and then again from line 430.

That probably also means that when you pick "readme" it squashes the changelog too, thanks to the call on line 430.

Reply:
Ah yes. I threw a `process.exit()` in the changeloggerSquash function. That's not my preferred method of doing this, but this file needs a lot more love and restructuring in a separate PR. Curse my early apprentice days.

Reply:
If it makes you feel any better, I often enough feel the same way looking at my old code 😀